File: app/src/main/java/com/a2017398956/nodesignmodeframework/exception/test/ResultInfoExceptionActivity.java
package com.a2017398956.nodesignmodeframework.exception.test;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.a2017398956.nodesignmodeframework.R;
import com.nfl.libraryoflibrary.utils.ToastTool;

/**
 * Created by fuli.niu
 */
public class ResultInfoExceptionActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result_info_exception);
    }

    public void exceptionTest(View view){
        Intent intent = new Intent(this , CreateResultInfoActivity.class) ;
//        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK) ;
        startActivityForResult(intent , 10);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        ToastTool.showShortToast("requestCode:" + requestCode + " | "
                + "resultCode:" + resultCode + " | "
        );
        super.onActivityResult(requestCode, resultCode, data);
        if(requestCode == RESULT_OK){
            if(resultCode == 10){
                ToastTool.showShortToast("没有发生异常");
            }
        }
    }
}

File: app/src/main/java/com/a2017398956/nodesignmodeframework/exception/test/ResultInfoExceptionActivity.kt
package com.a2017398956.nodesignmodeframework.exception.test

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import com.a2017398956.nodesignmodeframework.R
import com.a2017398956.nodesignmodeframework.databinding.ActivityResultInfoExceptionBinding
import com.nfl.libraryoflibrary.utils.ToastTool
import com.nfl.libraryoflibrary.utils.inflate

/**
 * Created by fuli.niu
 */
class ResultInfoExceptionActivity : Activity() {

    private val binding by inflate<ActivityResultInfoExceptionBinding>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding.btnStart.setOnClickListener {
            val intent = Intent(this, CreateResultInfoActivity::class.java)
            startActivityForResult(intent, 10)
        }
        binding.btnNewTask.setOnClickListener {
            val intent = Intent(this, CreateResultInfoActivity::class.java)
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            startActivityForResult(intent, 10)
        }
    }

    @SuppressLint("LongLogTag")
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        Log.d(TAG, "requestCode:$requestCode | resultCode:$resultCode")
        ToastTool.showLongToast("requestCode:$requestCode | resultCode is RESULT_OK? ${resultCode == RESULT_OK}")
    }

    override fun onDestroy() {
        ToastTool.showShortToast("ResultInfoExceptionActivity onDestroy")
        super.onDestroy()
    }

    companion object {
        const val TAG = "ResultInfoExceptionActivity"
    }
}
